break/clusterABCD.md

```diff
@@ -0,0 +1,333 @@
+
+Use the same `select_dtypes()` method to create a dataframe with the numerical and ordinal features. Store this dataframe in the variable `num_df`.
+
+<details>
+<summary>Hint</summary>
+You can supply a list to `select_dtypes()` to select both the "int64" and "float64" types.
+</details>
+
+```python
+# Insert code to create `num_df` from only the numerical and ordinal features of `subset_df`.
+```
+
+Use the `info()` method to confirm that we selected only the features we want.
+
+```python
+num_df.info()
+```
+
+The estimable [scikit-learn library](https://sklearn.org/) provides many common utility functions for scaling and standardizing data. We will use the [`MinMaxScaler`](https://sklearn.org/modules/generated/sklearn.preprocessing.MinMaxScaler.html#sklearn.preprocessing.MinMaxScaler) to scale our numerical features to lie between zero and one.
+
+```python
+# Scale standardisation of numerical values
+minmax = MinMaxScaler(feature_range=(0, 1))
+
+num_features = pd.DataFrame(
+    minmax.fit_transform(num_df),
+    index=num_df.index,
+    columns=num_df.columns
+)
+```
+
+And then concatenate our categorical and numerical features into a new dataframe called `df_scaled`.
+
+```python
+df_scaled = pd.concat([num_features, cat_features], axis="columns")
+
+# Confirm that all data lies between zero and one.
+print(f"Data min, max: {df_scaled.min().min(), df_scaled.max().max()}")
+```
+
+### Choosing an appropriate distance metric
+
+Finally, we need to provide some notion of distance between different points in our dataset. ***Different distances are appropriate for different datatypes.*** For example, it is common to use the Euclidean distance or Manhattan distance for numerical datatypes and dice distance for categorical variables. For those, you could rely on the scipy function [`scipy.spatial.distance.pdist`](https://docs.scipy.org/doc/scipy/reference/generated/scipy.spatial.distance.pdist.html#scipy.spatial.distance.pdist) to generate the distance matrices.
+
+Since we have a mix of categorical and numerical features, we will use the [Gower's distance](https://www.jstor.org/stable/2528823?seq=1), which was proposed in 1971 as a measure of dissimilarity with mixed types of variables.
+
+We could write our own Python function to compute the Gower's distance, but one should always ask ***"is there a pre-existing package that will do this for me?"*** And of course, the answer is YES. There is a package called `gower` that will compute the Gower's distance matrix for us. It isn't available on the jupyterhub by default (and you might not have it on your local system).
+
+But we can install software without ever leaving the jupyter notebook. You can execute Terminal commands in the notebook cells by prepending an exclamation point/bang(!) to the beginning of the command. Let's install the gower package.
+
+```python
+!pip install gower
```

Comment:
just a note that this will run each time, and that's necessary unless people are using custom environments.

Reply:
Yep, I thought it might be useful for them to know how to install `gower` or whatever other package they want. On the other hand, I'm happy to PR against the reprohub to add `gower` to the Dockerfile. I'm assuming that would go [here](https://github.com/ABCD-ReproNim/reprohub/blob/928baf013b8de861ec653877dc4c9397ddc91c8d/docker/Dockerfile#L52), correct? Either way is fine with me.

Reply:
We discussed this issue in the Week 8 meeting. We decided its fine for it to run each time in this context, but if `gower` is needed for more general purposes during Project Week we will make sure to have it already installed on the Hub.